src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -65,6 +65,61 @@ class LogicValueActionSerializer(serializers.Serializer):
     )
 
 
+class SynchronizeChildrenVariableMappingSerializer(serializers.Serializer):
+    property = serializers.CharField(
+        required=True, label=_("Property that needs mapping.")
+    )
+    component_key = FormioVariableKeyField(
+        required=True,
+        label=_("Key of the editgrid's child component."),
+    )
+
+
+class SynchronizeChildrenActionConfigSerializer(serializers.Serializer):
+    source_variable = FormioVariableKeyField(
+        required=True,
+        allow_blank=False,
+        label=_("Key of the form variable that will be used as the source."),
+    )
+    destination_variable = FormioVariableKeyField(
+        required=True,
+        allow_blank=False,
+        label=_("Key of the form variable that will be used as the destination."),
+    )
+    children_mappings = SynchronizeChildrenVariableMappingSerializer(
+        many=True,
+        label=_("Synchronize children mapping"),
+    )
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+        mappings = attrs.get("children_mappings")
+
+        if not mappings:
+            raise serializers.ValidationError(
+                {"children_mappings": _("At least one mapping is required.")}
+            )
+
+        already_mapped = []
+        for mapping in mappings:
+            if mapping["component_key"] in already_mapped:
+                raise serializers.ValidationError(
+                    {
+                        "children_mappings": _(
+                            "A variable cannot be mapped multiple times."
+                        )
+                    }
+                )
+
+            already_mapped.append(mapping["component_key"])
```

Comment:
```py
from collections import Counter

counts = Counter(m["component_key"] for m in mappings)
duplicates = [key for key, count in counter.items() if count > 1]
if duplicates:
    raise ...   
```
and then you can also report in the validation error *which* keys are mapped multile times